### PR TITLE
Fixes #132 - medical_appointment: Makes date, time, and duration required

### LINKED ISSUE #132: medical_appointment: Error when you leave a non-required field blank
medical_appointment threw an error if you entered an appointment without a date and time, but the date and time field did not appear to be required.

The error was:

Error while validating constraint
unsupported operand type(s) for +: 'NoneType' and 'datetime.timedelta'

### PR DESCRIPTION
**Pull request by kellylougheed**
- Makes the date/time of the appointment and the duration required fields so that they don't throw the errors noted in #132 
- Uses decimal precision in the duration

### CODE REVIEW

**Comment by lasley** [comment]
@kellylougheed - please see #71

**Comment by kellylougheed** [comment]
@lasley do you want me to expand this PR to also implement Kanban stages for medical_appointment?

**Comment by lasley** [comment]
Oops sorry #171

**Comment by kellylougheed** [comment]
@lasley OK, that looks awesome. Should I do a forward port of that PR and just throw in the required fields? Or would it be easier to include the required fields in that PR? Or has the issue with those fields been fixed by the improvements?

**Comment by lasley** [comment]
Let's include the required fields in that PR

**Comment by kellylougheed** [comment]
@lasley OK, should I push to that PR myself or leave it to @obulkin?

**Comment by lasley** [comment]
You can just push the fix yourself

**Comment by kellylougheed** [comment]
Done @lasley -- although when I pulled the branch, it created an extra "merge" commit. Hopefully it will be squashed?

**Comment by lasley** [comment]
You added a lot more than just a merge commit. There are a lot more commits in that PR now, most of which don't involve the PR.

**Comment by obulkin** [comment]
K. I just cleaned up #171 with a force push :jedi:. Maybe we should rebase this branch onto ``feature/10.0/LABS-379-deprecate-medical_base_history`` and make this PR dependent on #171?

**Comment by lasley** [comment]
Sounds like a plan @obulkin

**Comment by obulkin** [comment]
@kellylougheed - #171 has been merged now, so you can resolve the conflicts and get this ready for review.

**Comment by kellylougheed** [comment]
Thanks @obulkin!

File: medical_appointment/models/medical_appointment.py
Comment (lasley) [inline medical_appointment/models/medical_appointment.py]: Add a `digits`, with a decimal precision specific for appointment times.

https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/addons/l10n_fr_hr_payroll/models/l10n_fr_hr_payroll.py#L11

https://github.com/odoo/odoo/blob/fc2e80cb4bcc450762c7ac5cb82a3e2d88062b38/addons/hr_payroll/data/hr_payroll_data.xml#L82

**Comment by kellylougheed** [comment]
Decimal precision added @lasley

File: medical_appointment/data/medical_appointment_data.xml
Comment (lasley) [inline medical_appointment/data/medical_appointment_data.xml]: Essentially all files need to be split by model, the same way we do with actual models. See [contribution docs, file naming](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming)

**Comment by kellylougheed** [comment]
Created a new file for the decimal precision data @lasley

File: medical_appointment/data/decimal_precision_data.xml
Comment (lasley) [inline medical_appointment/data/decimal_precision_data.xml:2]: Too many newlines

File: medical_appointment/data/decimal_precision_data.xml
Comment (lasley) [inline medical_appointment/data/decimal_precision_data.xml]: order is id, model, other stuff

also id should be `model_name_something`

**Review by lasley — CHANGES_REQUESTED** [review]
The XML needs to be declared in manifest

**Comment by kellylougheed** [comment]
File changed @lasley

**Pull request merged**
